src/grt/src/GlobalRouter.cpp

```diff
@@ -3359,6 +3418,71 @@ void GlobalRouter::reportLayerWireLengths()
   }
 }
 
+void GlobalRouter::reportNetWireLength(odb::dbNet* net,
+                                       bool global_route,
+                                       bool detailed_route,
+                                       bool verbose)
+{
+  block_ = db_->getChip()->getBlock();
+  if (global_route) {
+    int wl = computeNetWirelength(net);
+    logger_->info(GRT,
+                  237,
+                  "Net {} global route wire length: {}u",
+                  net->getName(),
+                  dbuToMicrons(wl));
+
+    if (verbose) {
+      reportNetLayerWirelengths(net);
+    }
+  }
+
+  if (detailed_route) {
+    odb::dbWire* wire = net->getWire();
+
+    if (wire == nullptr)
+      logger_->warn(GRT, 239, "Net {} does not have routes.", net->getName());
+
+    int64_t wl = wire->getLength();
+    logger_->info(GRT,
+                  240,
+                  "Net {} detailed route wire length: {}u",
+                  net->getName(),
+                  dbuToMicrons(wl));
+
+    if (verbose) {
+      reportNetDetailedRouteWL(wire);
+    }
+  }
+}
+
+void GlobalRouter::reportNetDetailedRouteWL(odb::dbWire* wire)
+{
+  std::vector<int64_t> lengths;
+  lengths.resize(db_->getTech()->getRoutingLayerCount() + 1);
+  odb::dbWireShapeItr shapes;
+  odb::dbShape s;
+  int tplen;
+  for (shapes.begin(wire); shapes.next(s);) {
+    if (!s.isVia()) {
+      tplen = s.getDX() - s.getDY();
+      if (tplen < 0)
+        tplen = -tplen;
```

Comment:
I don't get why it is s.getDX() **-** s.getDY() and not s.getDX() **+** s.getDY()

Reply:
I'm doing it in the same way `dbWire::getLength` computes the total wire length. You can check here: https://github.com/eder-matheus/OpenROAD/blob/report_wire_length/src/odb/src/db/dbWire.cpp#L939

Reply:
I guess they are trying to subtract the half width extension on the wire segment but your segments don't have such an extension.  That code is also wrong for a non-default extension but that is another issue.

Reply:
I'm using this to compute the per-layer wire length from the detailed routing. Why they don't have this extension since I'm looking at the same data structs?

Reply:
You are correct, I was thinking of the grt case.  Really this should be moved to a dbShape::getLength() method which both you and dbWire::getLength() could call.

Reply:
I've added it.